src/core/BUILD.gn

```diff
@@ -743,6 +743,7 @@ openthread_core_files = [
   "utils/slaac_address.hpp",
   "utils/srp_client_buffers.cpp",
   "utils/srp_client_buffers.hpp",
+  "utils/static-counter.hpp",
```

Comment:
nit: the other files here use `_` rather than `-` as a separator. Should we be consistent?